File: core/src/main/java/me/devsnox/planetsystem/core/world/VoidWorldGenerator.java
package me.devsnox.planetsystem.core.world;

import org.bukkit.World;
import org.bukkit.generator.BlockPopulator;
import org.bukkit.generator.ChunkGenerator;

import java.util.Collections;
import java.util.List;
import java.util.Random;

public class VoidWorldGenerator extends ChunkGenerator {

    @Override
    public List<BlockPopulator> getDefaultPopulators(final World world) {
        return Collections.emptyList();
    }

    @Override
    public boolean canSpawn(final World world, final int x, final int z) {
        return true;
    }

    @Override
    @SuppressWarnings("deprecation")
    public byte[] generate(final World world, final Random rand, final int chunkx, final int chunkz) {
        return new byte[32768];
    }
}

File: core/src/main/kotlin/de/astride/planetsystem/core/world/VoidWorldGenerator.kt
/*
 * © Copyright - Astride UG (haftungsbeschränkt) 2018 - 2019.
 */

package de.astride.planetsystem.core.world

import org.bukkit.World
import org.bukkit.WorldCreator
import org.bukkit.WorldType
import org.bukkit.generator.BlockPopulator
import org.bukkit.generator.ChunkGenerator
import java.util.*

class VoidWorldGenerator : ChunkGenerator() {

    override fun getDefaultPopulators(world: World?): List<BlockPopulator> = emptyList()

    override fun canSpawn(world: World, x: Int, z: Int): Boolean = true

    @Suppress("OverridingDeprecatedMember")
    override fun generate(world: World?, rand: Random?, chunkx: Int, chunkz: Int) = ByteArray(32768)

}

fun generateVoidWorld(name: String): World = WorldCreator.name(name).type(WorldType.NORMAL)
    .generateStructures(false)
    .environment(World.Environment.NORMAL)
    .generator(VoidWorldGenerator())
    .createWorld()